test/cpu_adaptiveavgpool.hpp

```diff
@@ -0,0 +1,311 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_CPU_AVGPOOL_HPP
+#define GUARD_CPU_AVGPOOL_HPP
+
+#include "tensor_holder.hpp"
+#include <miopen/tensor_view_utils.hpp>
+
+template <class T>
+void cpu_adaptiveavgpool_forward_1d(
+    tensor<T> input, tensor<T>& output, size_t N, size_t C, size_t H, size_t OH)
+{
+    auto dims  = input.desc.GetLengths();
+    auto numel = output.desc.GetElementSize();
+
+    auto input_tv  = miopen::get_inner_expanded_tv<3>(input.desc);
+    auto output_tv = miopen::get_inner_expanded_tv<3>(output.desc);
+
+    for(size_t gid = 0; gid < numel; gid++)
+    {
+        size_t nc = gid / OH, oh = gid % OH;
+        size_t n = nc / C, c = nc % C;
+
+        if(n >= N)
+            return;
+
+        size_t h  = static_cast<size_t>(std::floor(static_cast<float>(oh * H) / OH));
+        size_t kh = static_cast<size_t>(std::ceil(static_cast<float>((oh + 1) * H) / OH)) - h;
+
+        float sum = 0;
+        for(size_t ih = h; ih < (h + kh); ++ih)
+        {
+            sum += static_cast<float>(input[input_tv.get_tensor_view_idx({n, c, ih})]);
+        }
+
+        output[output_tv.get_tensor_view_idx({n, c, oh})] = static_cast<T>(sum / kh);
+    }
+}
+
+template <class T>
+void cpu_adaptiveavgpool_forward_2d(tensor<T> input,
+                                    tensor<T>& output,
+                                    size_t N,
+                                    size_t C,
+                                    size_t H,
+                                    size_t W,
+                                    size_t OH,
+                                    size_t OW)
+{
+    auto dims  = input.desc.GetLengths();
+    auto numel = output.desc.GetElementSize();
+
+    auto input_tv  = miopen::get_inner_expanded_tv<4>(input.desc);
+    auto output_tv = miopen::get_inner_expanded_tv<4>(output.desc);
+
+    for(size_t gid = 0; gid < numel; gid++)
```

Comment:
You should use `par_ford` to perform parallel calculation, for better performance.